crates/types/src/journal_v2/mod.rs

```diff
@@ -78,6 +78,51 @@ impl fmt::Display for EntryType {
     }
 }
 
+// Avoid to_string() allocations when handling entries.
+impl EntryType {
+    pub fn as_static_str(&self) -> &'static str {
```

Comment:
nit: we could make it an impl Into<&'static str> instead of a method as_static_str as that is how strum does it